symphony/lib/toolkit/class.datasource.php

```diff
@@ -241,6 +241,45 @@ public function emptyXMLSet(XMLElement $xml = null)
     }
 
     /**
+     * Construct an XML fragment that describes a pagination structure.
+     * Do not call as a static function, since this is only temporary.
+     *
+     * @since Symphony 2.7.0
+     * @param array $infos (optional)
+     * @param integer $infos.$total-entries (optional)
+     *  the total number of entries that this structure is paginating. this
+     *  defaults to 0.
+     * @param integer $infos.$total-pages (optional)
+     *  the total number of pages within the pagination structure. this defaults
+     *  to 0.
+     * @param integer $infos.$entries-per-page (optional)
+     *  the number of entries per page. this defaults to 1.
+     * @param integer $infos.$current-page (optional)
+     *  the current page within the total number of pages within this pagination
+     *  structure. this defaults to 1.
+     * @return XMLElement
+     *  the constructed XML fragment.
+     */
+    public static function buildPaginationElement(array $infos = array())
```

Comment:
In order to make function signature resilient to changes, its a good practice to try to replace parameter with defaults values with a "settings bags".

You can then

1. Specify the 3rd option only, and keep the default intact.
2. Add a new option without worrying about signatures/API compat
3. Extension could ship there could with new option and it should not break on old install!

It only needs to be documented

Those parameters was the perfect example of this code smell and I did wanted to try it out.